pkg/agent/cniserver/server.go

```diff
@@ -373,6 +363,9 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 		// Rollback to delete configurations once ADD is failure.
 		if !success {
 			klog.Warningf("CmdAdd has failed, and try to rollback")
+			if !util.IsInfraContainer(netNS) {
```

Comment:
When this could happen? Why we do not rollback for non-infra containers?

Reply:
> When this could happen?
For workload container, `CmdAdd` may fail due to endpointHotAttach fail or endpoint not exist.

> Why we do not rollback for non-infra containers?
 When `CmdAdd` failed on workload container. No additional action is needed, so we can return error directly.

Done.

Reply:
Question - in this case do we need to set success to false for workload container? And seems you should at least remove or revise the log in line 365?

Reply:
Yes we stiil need to set success to false for workload container.

Update the log in line 365. Done.